### PR TITLE
Issue handler/reporter to report on duplicate ids

### LINKED ISSUE #6: filealreadyexist exception when run on quarkus guides
i assume quarkus current guides breaks some assumptions but might be good to handle the error better.

here is what happens when run on quarkus/asciidoc guides

```
 java -jar ~/code/work/asciidoc-splitter/target/asciidoc-splitter-1.0-SNAPSHOT.jar -s . -o ../result
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.headius.backport9.modules.Modules (file:/Users/max/code/work/asciidoc-splitter/target/asciidoc-splitter-1.0-SNAPSHOT.jar) to method sun.nio.ch.NativeThread.signal(long)
WARNING: Please consider reporting this to the maintainers of com.headius.backport9.modules.Modules
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
java.lang.RuntimeException: java.nio.file.FileAlreadyExistsException: /Users/max/code/work/quarkus/quarkus/docs/src/main/asciidoc/../result/modules/native-limitations.adoc
	at com.redhat.documentation.asciidoc.cli.ExtractionRunner.writeModules(ExtractionRunner.java:155)
	at com.redhat.documentation.asciidoc.cli.ExtractionRunner.call(ExtractionRunner.java:75)
	at com.redhat.documentation.asciidoc.cli.ExtractionRunner.call(ExtractionRunner.java:38)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1783)
	at picocli.CommandLine.access$900(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2150)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2144)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2108)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:1975)
	at picocli.CommandLine.execute(CommandLine.java:1904)
	at com.redhat.documentation.asciidoc.cli.ExtractionRunner.main(ExtractionRunner.java:56)
Caused by: java.nio.file.FileAlreadyExistsException: /Users/max/code/work/quarkus/quarkus/docs/src/main/asciidoc/../result/modules/native-limitations.adoc
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:94)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:215)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:370)
	at java.base/java.nio.file.Files.createFile(Files.java:647)
	at com.redhat.documentation.asciidoc.cli.ExtractionRunner.writeModules(ExtractionRunner.java:124)
	... 10 more
```

﻿

### PR DESCRIPTION
**Pull request by maxandersen**
when parsing quarkus current docs at tons of overlapping ids are found.
instead of ignoring or failing we now collect those issues found and
report them.

Fixes #6

### CODE REVIEW

File: src/main/java/com/redhat/documentation/asciidoc/cli/ExtractionRunner.java [hunk 1]
```diff
@@ -24,14 +21,13 @@
...
 import com.redhat.documentation.asciidoc.extraction.ExtractedModule;
-import com.redhat.documentation.asciidoc.extraction.SourceExtractor;
 import org.asciidoctor.Asciidoctor;
 import org.asciidoctor.OptionsBuilder;
 import org.asciidoctor.ast.Block;
 import org.asciidoctor.ast.Document;
 import org.asciidoctor.ast.Section;
-import org.asciidoctor.ast.StructuralNode;
 import org.asciidoctor.jruby.AsciiDocDirectoryWalker;
+import org.asciidoctor.jruby.ast.impl.BlockImpl;
```
Comment (LightGuard) [inline src/main/java/com/redhat/documentation/asciidoc/cli/ExtractionRunner.java]: We don't use this, and shouldn't be coding to the impl anyway.

File: src/main/java/com/redhat/documentation/asciidoc/cli/ExtractionRunner.java [hunk 2]
```diff
@@ -77,22 +74,51 @@ public Integer call() {
             writeAssemblies(config);
         }
 
+        int errors = 0;
+        for (Issue issue : this.issues) {
+            if(issue.isError()) {
+                errors++;
+            }
+        }
```
Comment (LightGuard) [inline src/main/java/com/redhat/documentation/asciidoc/cli/ExtractionRunner.java]: You could probably reduce this using a stream and a filter, probably easier to read.

File: src/main/java/com/redhat/documentation/asciidoc/cli/ExtractionRunner.java [hunk 3]
```diff
@@ -77,22 +74,51 @@ public Integer call() {
...
+                    }
+                }
+            } /* else if(node instanceof BlockImpl) {
+                Block block = ((Block)node);
+                if(!block.getSource().isBlank()) {
+                    addIssue(Issue.nonerror("Found Block that is non-empty: \n" + block.getSource(), node));
+                }
+            } else {
+                addIssue(Issue.nonerror("Found node(" + node.getClass().getSimpleName() + ") that is not a section or not at right level", node));
+            } */
```
Comment (LightGuard) [inline src/main/java/com/redhat/documentation/asciidoc/cli/ExtractionRunner.java]: If this is all commented out, do we need it in the PR?

**Comment by maxandersen** [comment]
fixed the issues mentioned and pushed update.

**Comment by LightGuard** [comment]
Thanks Max

**Pull request merged**